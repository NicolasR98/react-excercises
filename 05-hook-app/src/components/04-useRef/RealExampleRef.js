import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleRef = () => {
	const [show, setShow] = useState(true);

	return (
		<div>
			<h1>Real Example Ref</h1>
			<hr />

			{show && <MultipleCustomHooks />}

			<hr />

			<button
				className="btn btn-outline-primary"
				onClick={() => setShow(!show)}
			>
				Show/Hide
			</button>
		</div>
	);
};
