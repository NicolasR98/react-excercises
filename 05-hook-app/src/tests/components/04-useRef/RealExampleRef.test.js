import { shallow } from 'enzyme';
import React from 'react';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Tests on <RealExampleRef />', () => {
	const wrapper = shallow(<RealExampleRef />);

	test('should display correctly', () => {
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('MultipleCustomHooks').exists()).toBeFalsy();
	});

	test('should show <MultipleCustomHooks /> component when click', () => {
		wrapper.find('button').simulate('click');

		expect(wrapper.find('MultipleCustomHooks').exists()).toBeTruthy();
	});
});
