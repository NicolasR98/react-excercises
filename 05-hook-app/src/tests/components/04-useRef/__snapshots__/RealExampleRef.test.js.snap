// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <RealExampleRef /> should display correctly 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <hr />
  <button
    className="btn btn-outline-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
